### ISSUE TITLE
Consider adding credit for original code

### ISSUE THREAD

**Issue by MangosArentLiterature** [lambdcalculus/scs#2]
Large portions of this repository appear to be copied and based off code I originally wrote as part of [Athena](https://github.com/MangosArentLiterature/Athena). While I don't mind my code being used in such a manner, so long as the terms of Athena's license are adhered to, I would still personally appreciate some form of credit or acknowledgement for this fact/the code that was originally my own, particularly if this project were to ever be made available for use.

**Comment by lambdcalculus** [comment]
Reasonable! I did use Athena as a starting point for base features and structure (though it was not my intent to copy - I'd like to hope some implementation differences make that clear), so acknowledgement will absolutely be given once the project has a README. If you'd like, it can also be included in files/parts where you feel credit is particularly due.

**Comment by MangosArentLiterature** [comment]
An acknowledgement and a link back to Athena would be appreciated. There were a few areas I noticed that were almost identical to code I wrote for Athena (such as my implementation of IPIDs), but I don't see much need to go and specifically credit every instance of this so long as it's made clear that this project was based off Athena and uses code adapted from it.

**Comment by lambdcalculus** [comment]
Yeah, acknowledgement and link is what I was going to go for, no worries there.

And I'll say that a lot of code that looks similar came from looking at how both Athena and Akashi do things, trying to come up with a better way, and ultimately picking an approach from either or mixing both. IPIDs is an example: both Akashi and Athena use a MD5 hash, with Athena using base64 to make it shorter and Akashi taking the first 32 bits. I combine the approaches by also base64'ing the MD5 hash, and taking the first 36 bits, making for 6-character IDs that are quicker to type but also provide better randomness than Akashi.

Like many other things that ended up the same, I spent time looking for other approaches to shorten MD5 hashes, and base64 was what came up most often with few reasonable alternatives, so I went for it like Athena did. Rest assured it was not a decision based on copy-paste, and all the similar code went through a similar process.
